ibis/backends/duckdb/tests/test_register.py

```diff
@@ -108,6 +108,36 @@ def test_read_geo_to_geopandas(con, data_dir):
     assert isinstance(gdf, gpd.GeoDataFrame)
 
 
+@pytest.mark.xfail(
+    LINUX and SANDBOXED,
+    reason="nix on linux cannot download duckdb extensions or data due to sandboxing",
+)
+def test_read_geo_from_url(tmpdir):
+    # httpfs is needed to read from url on nightly spatial extension
+    gpd = pytest.importorskip("geopandas")
+
+    path = str(tmpdir.join("test_read_geo_url.ddb"))
+
+    config = {"extension_directory": str(tmpdir.join("extensions"))}
+
+    con = ibis.duckdb.connect(path, **config)
+    # TODO(ncclementi): remove force install when spatial is updated
+    # see https://github.com/duckdb/duckdb_spatial/issues/210
+    con.raw_sql(
+        """\
+        FORCE INSTALL spatial FROM 'http://nightly-extensions.duckdb.org';
+        LOAD spatial;
+        """
+    )
+
+    url = "https://raw.githubusercontent.com/ibis-project/testing-data/master/geojson/zones.geojson"
+
+    t = con.read_geo(url)
+
+    gdf = t.head().to_pandas()
+    assert isinstance(gdf, gpd.GeoDataFrame)
+
+
```

Comment:
so rather than having the test suite load this file from the internet, I think we can test that, given a string that begins with `https`, that the `spatial` and `httpfs` extensions are loaded.  That way we're only testing that Ibis is doing what we expect for a given user behavior.

That could look something like:
```suggestion
def test_read_geo_from_url(con, monkeypatch):
    loaded_exts = []
    monkeypatch.setattr(con, "_load_extensions", lambda x, **kw: loaded_exts.extend(x))

    with pytest.raises(sa.exc.OperationalError):
        # The read will fail
        con.read_geo("https://...")

    assert "spatial" in loaded_exts
    assert "httpfs" in loaded_exts

```